core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -866,11 +866,14 @@ class BytesToString extends org.apache.spark.api.java.function.Function[Array[By
 }
 
 /**
- * Internal class that acts as an `AccumulatorParam` for Python accumulators. Inside, it
+ * Internal class that acts as an `AccumulatorV2` for Python accumulators. Inside, it
  * collects a list of pickled strings that we pass to Python through a socket.
  */
-private class PythonAccumulatorParam(@transient private val serverHost: String, serverPort: Int)
-  extends AccumulatorParam[JList[Array[Byte]]] {
+private[spark] class PythonAccumulatorV2(@transient private val serverHost: String, serverPort: Int)
+  extends AccumulatorV2[JList[Array[Byte]], JList[Array[Byte]]] {
```

Comment:
Yeah, `CollectionAccumulator` would need you to add `Array[Byte]` to get out `List[Array[Byte]]`. But actually, the only thing that's added is a singleton list of one `Array[Byte]` so the usage is already expecting to add one element, not a list.

Then you don't need a custom implementation at all. Just `CollectionAccumulator[Array[Byte]]`?